cli/azd/pkg/infra/azure_resource_types.go

```diff
@@ -1,14 +1,76 @@
+	AzureResourceTypeAppInsightComponent     AzureResourceType = "Microsoft.Insights/components"
+	AzureResourceTypeLogAnalyticsWorkspace   AzureResourceType = "Microsoft.OperationalInsights/workspaces"
+	AzureResourceTypeWebSite                 AzureResourceType = "Microsoft.Web/sites"
+	AzureResourceTypeStaticWebSite           AzureResourceType = "Microsoft.Web/staticSites"
+	AzureResourceTypeServicePlan             AzureResourceType = "Microsoft.Web/serverfarms"
+	AzureResourceTypeSqlDatabase             AzureResourceType = "Microsoft.Sql/servers"
+	AzureResourceTypeCosmosDb                AzureResourceType = "Microsoft.DocumentDB/databaseAccounts"
+	AzureResourceTypeContainerApp            AzureResourceType = "Microsoft.App/containerApps"
+	AzureResourceTypeContainerAppEnvironment AzureResourceType = "Microsoft.App/managedEnvironments"
 )
+
+const resourceLevelSeparator = "/"
+
+// GetResourceTypeDisplayName retrieves the display name for the given resource type.
+// If the display name was not found for the given resource type, an empty string is returned instead.
+func GetResourceTypeDisplayName(resourceType AzureResourceType) string {
+	// Azure Resource Manager does not offer an API for obtaining display name for resource types.
+	// Display names for Azure resource types in Azure Portal are encoded in UX definition files instead.
+	// As a result, we provide static translations for known resources below. These are obtained from the Azure Portal.
+	switch resourceType {
+	case AzureResourceTypeResourceGroup:
+		return "Resource group"
+	case AzureResourceTypeStorageAccount:
+		return "Storage account"
+	case AzureResourceTypeKeyVault:
+		return "Key vault"
+	case AzureResourceTypePortalDashboard:
+		return "Portal dashboard"
+	case AzureResourceTypeAppInsightComponent:
+		return "Application Insights"
+	case AzureResourceTypeLogAnalyticsWorkspace:
+		return "Log Analytics workspace"
+	case AzureResourceTypeWebSite:
+		return "Web App"
+	case AzureResourceTypeStaticWebSite:
+		return "Static Web App"
+	case AzureResourceTypeContainerApp:
+		return "Container App"
+	case AzureResourceTypeContainerAppEnvironment:
+		return "Container Apps Environment"
+	case AzureResourceTypeServicePlan:
+		return "App Service plan"
+	case AzureResourceTypeCosmosDb:
+		return "Azure Cosmos DB"
+	}
+
+	return ""
+}
+
+// IsTopLevelResourceType returns true if the resource type is a top-level resource type, otherwise false.
+// A top-level resource type is of the format of: {ResourceProvider}/{TopLevelResourceType}, i.e. Microsoft.DocumentDB/databaseAccounts
+func IsTopLevelResourceType(resourceType AzureResourceType) bool {
+	resType := string(resourceType)
+	firstIndex := strings.Index(resType, resourceLevelSeparator)
+
+	if firstIndex == -1 ||
+		firstIndex == 0 ||
+		firstIndex == len(resType)-1 {
+		return false
+	}
+
+	secondIndex := strings.Index(resType[firstIndex+1:], resourceLevelSeparator)
```

Comment:
Maybe just use strings.Contains here (and return the negation of the value, since it's not a top level resource if there is another `/` in the string).